File: LeetCode/[E] - 724. Find Pivot Index/solution.go
package main

func pivotIndex(nums []int) int {
	sumNum := 0
	for _, el := range nums {
		sumNum += el
	}
	sumCurr := 0
	for i := 0; i < len(nums); i++ {
		if sumNum-sumCurr-nums[i] == sumCurr {
			return i
		}
		sumCurr += nums[i]
	}
	return -1
}

File: LeetCode/[E] - 724. Find Pivot Index/solution.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        sum_nums = sum(nums)
        sum_curr = 0
        for i in range(len(nums)):
            if sum_nums - sum_curr - nums[i] == sum_curr:
                return i
            sum_curr += nums[i]
        return -1
